src/components/BadgeManager.tsx

```diff
@@ -0,0 +1,110 @@
+import { lastReadTimeLocalForage } from '@/hooks/useLastReadMessageId'
+import useWrapInRef from '@/hooks/useWrapInRef'
+import { followedIdsStorage, useMyAccount } from '@/stores/my-account'
+import { getSquidUrl } from '@/utils/env/client'
+import { gql, GraphQLClient } from 'graphql-request'
+import { useEffect } from 'react'
+
+export default function BadgeManager() {
+  const myAddress = useMyAccount((state) => state.address)
+  const myAddressRef = useWrapInRef(myAddress)
+
+  useEffect(() => {
+    function listener() {
+      if (document.visibilityState === 'hidden') syncBadge(myAddressRef.current)
+      else clearBadge()
+    }
+    document.addEventListener('visibilitychange', listener)
+    return () => document.removeEventListener('visibilitychange', listener)
+  }, [myAddressRef])
+
+  return null
+}
+
+async function syncBadge(address: string | null) {
+  const unreadCount = await getUnreadCount(address)
+  if (!unreadCount) {
+    await clearBadge()
+  } else {
+    await setBadge(unreadCount)
+  }
+}
+
+async function getUnreadCount(address: string | null) {
+  const squidUrl = getSquidUrl()
+  if (!squidUrl) return 0
+
+  let chatIdsToFetch = ['754', '7465']
```

Comment:
extract to config, hard to find the ids when someone want to add new chat

Reply:
Updated, added to chat constants